packages/blockchain-wallet-v4-frontend/src/components/Card/ExchangePromo/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { Experiments } from '@core/network/api/experiments/types'
+import { actions, selectors } from 'data'
+import { ExchangeAuthOriginType } from 'data/types'
+
+import ExchangePromo from './ExchangePromo'
+
+// local storage keys for dismiss stuff
+const DATE_DISSMISSED_KEY = 'exchange-promo-dismiss-date'
+const DISSMISSED_KEY = 'exchange-promo-dismiss-forever'
+const A_WEEK = 7 * 24 * 60 * 60 * 1000
+
+// From experiments API
+const isExperimentOn = (experiments: Experiments) => {
+  if (!Object.keys(experiments).length) return false
+  return experiments.walletAwarenessPrompt === 1
+}
+
+const getDismissedDate = (): number | null => {
+  const rawDate = localStorage.getItem(DATE_DISSMISSED_KEY)
+  if (!rawDate) return null
+  return parseInt(rawDate, 10)
+}
+
+const wasDismissed = () => {
+  // Card dismissed twice so gone forever
+  const dismissed = localStorage.getItem(DISSMISSED_KEY)
+  if (dismissed) return true
+
+  // Card dismissed once, hide for a week
+  const dateDismissed = getDismissedDate()
+  if (dateDismissed && dateDismissed + A_WEEK > +new Date()) {
+    return true
+  }
+  return false
+}
+
+const onDismiss = () => {
+  const dateDismissed = getDismissedDate()
+  if (dateDismissed) {
+    // dissmissed twice now
+    localStorage.setItem(DISSMISSED_KEY, 'true')
+  } else {
+    localStorage.setItem(DATE_DISSMISSED_KEY, String(+new Date()))
+  }
+}
+
+const ExchangePromoContainer = () => {
+  const dispatch = useDispatch()
+  useEffect(() => {
+    dispatch(actions.experiments.fetch())
+  }, [dispatch])
+
+  const experimentsState = useSelector(selectors.experiments.get)
+  const experiments = experimentsState.getOrElse({})
```

Comment:
Is this ok? I started with the `.cata` approach but then I realised I only need to render something when this is on the store.